### PR TITLE
hw02

### PR DESCRIPTION
**Pull request by AlenaVyrodova**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -48,6 +57,20 @@
...
+        <dependency>
+            <groupId>org.springframework</groupId>
+            <artifactId>spring-web</artifactId>
+            <version>6.0.12</version>
+        </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.core</groupId>
+            <artifactId>jackson-annotations</artifactId>
+            <version>2.15.2</version>
+        </dependency>
```
Comment (sokimaaa) [inline pom.xml]: redundant dependencies should be removed.

File: src/main/java/com/example/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.web.bind.annotation.RestController;
+
+@RequiredArgsConstructor
+@RestController
+@RequestMapping(value = "/books")
+public class BookController {
+    private final BookService bookService;
+
+    @GetMapping
+    public List getAll() {
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/controller/BookController.java]: Never use raw type param. 

Should be parametrized with concrete type.

File: src/main/java/com/example/bookstore/service/BookServiceImpl.java [hunk 3]
```diff
@@ -1,28 +1,39 @@
...
+                .stream().map(bookMapper::toDto).toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(
+                () -> new EntityNotFoundException("Can't find book by id" + id));
+        return bookMapper.toDto(book);
```
Comment (sokimaaa) [inline src/main/java/com/example/bookstore/service/BookServiceImpl.java]:
```suggestion
        return bookRepository.findById(id).map(bookMapper::toDto).orElseThrow(
                () -> new EntityNotFoundException("Can't find book by id" + id));
```

File: pom.xml [hunk 4]
```diff
@@ -12,17 +12,30 @@
     <artifactId>bookStore</artifactId>
     <version>0.0.1-SNAPSHOT</version>
     <name>bookStore</name>
+    <packaging>war</packaging>
```
Comment (Rommelua) [inline pom.xml:15]:
```suggestion
```

Comment (AlenaVyrodova) [reply pom.xml:15]: Thanks a lot!

**Pull request merged**